inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -16,255 +16,535 @@
-            return sublist;
-        }
+      throw std::invalid_argument("In-Lane Cruising does not support this maneuver type");
+    }
 
-        ROS_WARN_STREAM("18");
+    cav_msgs::LaneFollowingManeuver lane_following_maneuver = manuever.lane_following_maneuver;
 
-        // Get waypoints from nearest waypoint to time boundary
-        size_t index = std::get<1>(nearest_points[0]);
+    double starting_downtrack = lane_following_maneuver.start_dist;
+    if (first)
+    {
+      if (starting_downtrack > max_starting_downtrack)
+      {
+        starting_downtrack = max_starting_downtrack;
+      }
+      first = false;
+    }
 
-        ROS_WARN_STREAM("19");
-        if (index == waypoints.size() - 1) {
-            ROS_INFO_STREAM("Nearest point is final waypoint so it is being dropped");
-            return sublist;
-        }
+    ROS_DEBUG_STREAM("Used downtrack: " << starting_downtrack);
 
-        ROS_WARN_STREAM("20");
+    auto lanelets = wm->getLaneletsBetween(starting_downtrack, lane_following_maneuver.end_dist, true);
 
-        double total_time = 0.0;
-        size_t start_index = index + 1;
-        for(int i = start_index; i < waypoints.size(); ++i) // Iterate starting from the waypoint after nearest to ensure it is beyond the current vehicle position
-        {
-            sublist.push_back(waypoints[i]);
-            if(i == start_index)
-            {
-                ROS_WARN_STREAM("21");
-                continue;
-            }
-            ROS_WARN_STREAM("20");
-            double delta_x_square = pow(waypoints[i].pose.pose.position.x - waypoints[i - 1].pose.pose.position.x, 2);
-            double delta_y_square = pow(waypoints[i].pose.pose.position.y - waypoints[i - 1].pose.pose.position.y, 2);
-            //double delta_z_square = waypoints[i].pose.pose.position.z - waypoints[i - 1].pose.pose.position.z;
-            // Here we ignore z attribute because it is not used by Autoware
-            double delta_d = sqrt(delta_x_square + delta_y_square);
-            double average_v = 0.5 * (waypoints[i].twist.twist.linear.x + waypoints[i - 1].twist.twist.linear.x);
-            double delta_t = delta_d / average_v;
-            total_time += delta_t;
-            if(total_time >= time_span)
-            {
-                break;
-            }
-        }
-        return sublist;
+    ROS_DEBUG_STREAM("Maneuver");
+    std::vector<lanelet::ConstLanelet> lanelets_to_add;
+    for (auto l : lanelets)
+    {
+      ROS_DEBUG_STREAM("Lanelet ID: " << l.id());
+      if (visited_lanelets.find(l.id()) == visited_lanelets.end())
+      {
+        lanelets_to_add.push_back(l);
+        visited_lanelets.insert(l.id());
+      }
     }
 
-    std::vector<cav_msgs::TrajectoryPlanPoint> InLaneCruisingPlugin::post_process_traj_points(std::vector<cav_msgs::TrajectoryPlanPoint> trajectory)
+    lanelet::BasicLineString2d route_geometry = carma_wm::geometry::concatenate_lanelets(lanelets_to_add);
+
+    bool first = true;
```

Comment:
this var seems to have been declared again. Should we use different names to distinguish from the one above?